Explorer.Series.pow integer support

kimjoaoun opened the pull request:
Adds support for Integers on Series.pow()

The version in this PR supports only positive integers. 

This pull request, when completed, closes #105.

kimjoaoun commented:
Currently working on adding support for negative integers.

kimjoaoun commented:
Marked as ready for review since I don't think It'll be necessary to add support do negative integers, since not even Elixir's `Integer.pow/2` supports them.
You can use Explorer.Series.pow with a negative floating point number, as expected.

josevalim commented:
Yeah, the issue with supporting negative integers is that they return floats, so you have an api signature that may return integer or floats and that can be confusing. So if you have negative integers, you need to cast them to floats before.

josevalim commented on lib/explorer/polars_backend/series.ex:
```diff
@@ -235,9 +235,11 @@ defmodule Explorer.PolarsBackend.Series do
+  def pow(left, exponent) when is_float(exponent),
     do: Shared.apply_native(left, :s_pow, [exponent])
 
+  def pow(left, exponent) when is_integer(exponent),
```
```suggestion
  def pow(left, exponent) when is_integer(exponent) and exponent >= 0,
```

kimjoaoun replied:
I think this change would result in a exception if the user try to evaluate the following:

```Ex
iex> s = [2, 4, 6] |> Explorer.Series.from_list()
iex> Explorer.Series.pow(s, -3.0)
```

cigrainger commented on native/explorer/src/series.rs:
```diff
@@ -708,6 +708,13 @@ pub fn s_pow(data: ExSeries, exponent: f64) -> Result<ExSeries, ExplorerError> {
 }
 
+#[rustler::nif]
+pub fn s_int_pow(data: ExSeries, exponent: i64) -> Result<ExSeries, ExplorerError> {
```
You should actually be able to just use `u32` in the signature here for `exponent` and `rustler` should automatically marshall correctly.

cigrainger reviewed (commented):
Fantastic! Thanks for doing this so quickly. Might be worth updating `s_pow` to use the same pattern as Ritchie indicated `Series::pow` will be deprecated and it's just one less thing to do down the road. But perfectly fine not to as well :).

kimjoaoun commented:
@cigrainger It's my pleasure to help! I'm learning a lot in these contributions, it is being an amazing experience :')

I updated `s_pow` but I'm not an avid Rust programmer, so I don't know if I did so in the best implementation possible.

cigrainger reviewed (approved):
It looks good to me! Thanks again, this is great.

cigrainger commented:
Okay, one last request and then I think it's ready to merge -- could you please add an example with a float series and float exponent? Also, what happens if you have a float series and int exponent?

kimjoaoun commented:
> Okay, one last request and then I think it's ready to merge -- could you please add an example with a float series and float exponent? Also, what happens if you have a float series and int exponent?

I added the docs. Before these recent commits if you tried to run code with a float series and a int exponent it would break, but now I fixed this case.

cigrainger reviewed (approved):
Fantastic! Thanks again.
